docs/evolutionaryAlgorithm/README.md

```diff
@@ -0,0 +1,264 @@
+
+1. Download the data from open street map:
+
+```
+$ yarn node packages/chaire-lib-backend/lib/scripts/osm/1_downloadOsmDataToFiles.task.js
+yarn node v1.22.22
+Using .env file from ./.env
+Read server configuration from ./examples/config.js
+? Please select a data source for import (Use arrow keys)
+❯ villeray 
+  --New data source--
+? Please select a data source for import 
+Retrieving data from osm...
+Done retrieving data from osm...
+Writing osm data to runtime/imports/villeray/osmRawData.json
+Done writing osm data
+Retrieving data from osm...
+Done retrieving data from osm...
+Writing osm geojson data to runtime/imports/villeray/osmData.geojson
+Done writing osm geojson data
+```
+
+2. (optional) Enhance the polygon data. it adds the polygon area in sq meters, for polygons, it calculates the area and the floor area using the building:levels value and the calculated area, or it takes the existing floor area from the building:floor_area tag, which is supposed to be in sq meters.
+```
+yarn node packages/chaire-lib-backend/lib/scripts/osm/1b_prepareAndEnhanceOsmPolygonData.task.js
+? Please select a data source for import (Use arrow keys)
+❯ villeray 
+  --New data source--
+Running import in dir runtime/imports/villeray/
+Enhancing and saving polygon geojson data to file:  runtime/imports/villeray/osmPolygonsEnhanced.geojson
+Fetch buildings from osm geojson data...
+There are 16103 buildings in the osm geojson data.
+Fetch polygons and multipolygons from osm geojson data...
+=== validated and enhanced polygon 16970 ===
+Finished with 16971 polygons.
+```
+
+3. Import and process osm data to files. This will create 3 geojson files, one for points of interest, one for residential buildings and one for residential zones. This task will give a number of warnings on buildings and places that may have missing or incomplete data and for which an approximation is used. It is worth running this task in non-interactive mode and pipe the output to a file for later processing
+
+```
+yarn node packages/chaire-lib-backend/lib/scripts/osm/2_importAndProcessOsmDataToFiles.task.js --dataSourceId villeray > outputFile.txt
+```
+
+4. (optional) Assign weights to POIs. This will use the "Trip Generation Manual"'s data to give weight to POIs according to their category and the area occupied.
+
+```
+yarn node packages/chaire-lib-backend/lib/scripts/osm/2b_assignWeightsToPOIs.task.js
+```
+
+5. (optional?) Import and validate land role data. This will validate that the number of residences (flats) from openstreetmap matches the land role data and give warnings if it is not the case.
```

Comment:
Guessing with the "?" that you are not sure if it's required or not at this point?

Reply:
Je crois qu'on peut enlever le ? c,est optionnel. Ceci dit, chaque residence compte pour 1 et faut se fier aux flats d'OSM au lieu de les avoir du land role. SI on veut utiliser les lieux de résidences dans les calculs, c'est mieux d'avoir le land role. Je vais ajouter ça